src/coreclr/debug/daccess/dacdbiimplstackwalk.cpp

```diff
@@ -185,7 +189,23 @@ void DacDbiInterfaceImpl::SetStackWalkCurrentContext(VMPTR_Thread           vmTh
     // Allocate a context in DDImpl's memory space. DDImpl can't contain raw pointers back into
     // the client space since that may not marshal.
     T_CONTEXT * pContext2 = GetContextBufferFromHandle(pSFIHandle);
-    CopyMemory(pContext2, pContext, sizeof(*pContext));
+
+#ifdef FEATURE_INTERPRETER
+    PTR_Frame pTopFrame = vmThread.GetDacPtr()->GetFrame();
+
+    if ((pTopFrame != FRAME_TOP) && (pTopFrame->GetFrameIdentifier() == FrameIdentifier::InterpreterEntryFrame))
+    {
+        InterpreterEntryFrame *pEntryFrame = dac_cast<PTR_InterpreterEntryFrame>(pTopFrame);
+        PTR_InterpMethodContextFrame pTOSInterpMethodContextFrame = pEntryFrame->GetInterpMethodTopmostContextFrame();
+        SetIP(pContext2, (TADDR)pTOSInterpMethodContextFrame->ip);
+        SetSP(pContext2, dac_cast<TADDR>(pTOSInterpMethodContextFrame));
```

Comment:
Do you think that overloading SP to mean interpreter stack pointer when we are executing interpreter code is going to work well?

For example, it means that we won't be able to simple compare two stack pointers to determine relative position of two frames on the stack. We will have to walk the whole chain instead.

Reply:
Why? The SPs are still ordered correctly.

Reply:
Ah ok, I got confused by the reuse scheme.

Reply:
To explain in more detail, the SP is actually not pointing to the interpreter stack in the sense of the stack that interpreter uses for its locals and call arguments. It is a pointer to the `InterpMethodContextFrame` structures that are allocated on regular stack using `alloca` at each interpreter call (they are reused when the interpreter call returns and another one happens).